core/modules/catalog/components/ProductBundleOptions.ts

```diff
@@ -57,83 +38,69 @@ export const ProductBundleOptions = {
-        if (existingField && existingField.hasOwnProperty('value') && typeof existingField.value === 'object') {
-          opval = existingField.value
-        }
-      }
-      const fieldNameQty = _fieldName(option)[1]
-      const value = opval === null ? this.inputValues[fieldName] : opval.id
-      this.setBundleOptionValue({ optionId: option.option_id, optionQty: parseInt(this.inputValues[fieldNameQty]), optionSelections: [value] })
+    optionChanged({fieldName, option, qty, value}) {
+      if (!fieldName) return
+      this.setBundleOptionValue({ optionId: option.option_id, optionQty: parseInt(qty), optionSelections: [value.id] })
       this.$store.dispatch('product/setBundleOptions', { product: this.product, bundleOptions: this.$store.state.product.current_bundle_options }) // TODO: move it to "AddToCart"
-      this.selectedOptions[fieldName] = { value: (opval === null ? value : opval), qty: parseInt(this.inputValues[fieldNameQty]) }
-      if (this.validateField(option)) {
+      this.selectedOptions[fieldName] = {qty, value}
+      const valueId = value ? value.id : null
+      if (this.validateField(option, qty, valueId)) {
         this.$bus.$emit('product-after-bundleoptions', { product: this.product, option: option, optionValues: this.selectedOptions })
       }
     },
     isValid () {
       let isValid = true
-      this.validation.results.map((res) => { if (res.error) isValid = false })
+      this.validationResults.map((res) => { if (res.error) isValid = false })
       return isValid
     },
-    validateField (option) {
+    validateField (option, qty, optionId) {
       let result = true
+      let validationResult = { error: false, message: '' }
       for (let fieldName of _fieldName(option)) {
-        const validationRule = this.validation.rules[fieldName]
+        const validationRule = this.validationRules[fieldName]
         this.product.errors.custom_options = null
         if (validationRule) {
           const validator = this.$store.state.product.custom_options_validators[validationRule]
           if (typeof validator === 'function') {
-            const validationResult = validator(this['inputValues'][fieldName])
-            this.validation.results[fieldName] = validationResult
+            const quantityValidationResult = validator(qty)
+            if(quantityValidationResult.error) validationResult = quantityValidationResult
+            const optionValidationResult = validator(optionId)
+            if(optionValidationResult.error) validationResult = optionValidationResult
+            this.$set(this.validationResults, fieldName, validationResult)
             if (validationResult.error) {
-              this.product.errors['bundle_options_' + fieldName] = i18n.t('Please configure product bundle options and fix the validation errors')
+              this.product.errors['bundle_options_' + fieldName] = i18n.t('Please configure product custom options and fix the validation errors')
```

Comment:
shouldn't it be "bundle" instead of "custom" options?

Reply:
Translation file doesn't have "bundle" keypath, so i picked up closest entry match. With "bundle" string console shows i18 warnings:
`vue-i18n.esm.js:14 [vue-i18n] Cannot translate the value of keypath`

Reply:
Please just add this string to translation file (en-US)

Reply:
Sure, done :)